### ISSUE TITLE
`unreadCountPublisher` does not work until `currentUser` is called at least once

### ISSUE THREAD

**Issue by MatejBalantic** [GetStream/stream-chat-swift#1224]
## What did you do?
I tried to subscribe to `chatClient.currentUserController().unreadCountPublisher`

## What did you expect to happen?
I expected this signal will send new values every time the unread count changes

## What happened instead?
Only initial value of 0 was sent, no more values after that

## GetStream Environment
**GetStream Chat version:**
3.1.11

## Additional context

After looking through the SDK code I noticed that `startObservingIfNeeded` of the `_CurrentChatUserController` never gets called. 

I was able to work around the problem by calling `chatClient.currentUser` and discarding the results. That triggered the `startObservingIfNeeded()` call, which consequently started correctly sending values to the `unreadCountPublisher`

**Comment by VojtaStavik** [comment]
Thank you for reporting this @MatejBalantic 🙏

**Comment by VojtaStavik** [comment]
Tracked internally as CIS-970

**Comment by VojtaStavik** [comment]
Hi @MatejBalantic , I checked this issue and found out that this works as designed. The general pattern for the Combine wrappers is:
```
1. create the controller
2. update your UI with locally cached data available on the channel (for example controller.messages)
3. set up the Combine observers to receive updates
4. call synchronize
```
For values like messages or channels, the Combine stream propagates only diffs from the initial values before you call `synchronize`.

I agree that this doesn't translate well for simple values, like the unread count but it's an API design trade-off we had to make. The `sychronize` call can fail and if we do it implicitly, we'd have to change the publishers' error type from `Never` to a concrete error. It seemed like a better idea to let users handle the possible failure of the `sychronize` method outside of the Combine pipe.

Happy to hear your thoughts on this.

---

PS: The reason why accessing `currentUser` fixes the problem is more a coincidence due to the special nature of this particular controller. For example, the channel controller won't receive any server updates if you don't call `synchronize` on it.
